Zend/Zend.m4

```diff
@@ -119,415 +119,403 @@ AC_DEFUN([ZEND_CHECK_FLOAT_PRECISION],[
+      #define zend_isinf(a) isinf(a)
+      #elif defined(INFINITY)
+      /* Might not work, but is required by ISO C99 */
+      #define zend_isinf(a) (((a)==INFINITY)?1:0)
+      #elif defined(HAVE_FPCLASS)
+      #define zend_isinf(a) ((fpclass(a) == FP_PINF) || (fpclass(a) == FP_NINF))
+      #else
+      #define zend_isinf(a) 0
+      #endif
+
+      int main(int argc, char** argv)
+      {
+        return zend_isinf(HUGE_VAL) ? 0 : 1;
+      }
+    ]])],[
+      ac_cv_huge_val_inf=yes
+    ],[
+      ac_cv_huge_val_inf=no
+    ],[
+      ac_cv_huge_val_inf=yes
+    ])
+  ])
+  dnl This is the most probable fallback so we assume yes in case of cross compile.
+  if test "$ac_cv_huge_val_inf" = "yes"; then
+    AC_DEFINE([HAVE_HUGE_VAL_INF], 1, [whether HUGE_VAL == INF])
+  fi
 
-AC_CHECK_FUNCS(mremap)
+  dnl Check if HUGE_VAL + -HUGEVAL == NAN.
+  AC_CACHE_CHECK([whether HUGE_VAL + -HUGEVAL == NAN], ac_cv_huge_val_nan,[
+    AC_RUN_IFELSE([AC_LANG_SOURCE([[
+      #include <math.h>
+      #include <stdlib.h>
+
+      #ifdef HAVE_ISNAN
+      #define zend_isnan(a) isnan(a)
+      #elif defined(HAVE_FPCLASS)
+      #define zend_isnan(a) ((fpclass(a) == FP_SNAN) || (fpclass(a) == FP_QNAN))
+      #else
+      #define zend_isnan(a) 0
+      #endif
+
+      int main(int argc, char** argv)
+      {
+      #if defined(__sparc__) && !(__GNUC__ >= 3)
+        /* prevent bug #27830 */
+        return 1;
+      #else
+        return zend_isnan(HUGE_VAL + -HUGE_VAL) ? 0 : 1;
+      #endif
+      }
+    ]])],[
+      ac_cv_huge_val_nan=yes
+    ],[
+      ac_cv_huge_val_nan=no
+    ],[
+      ac_cv_huge_val_nan=yes
+    ])
+  ])
+  dnl This is the most probable fallback so we assume yes in case of cross compile.
+  if test "$ac_cv_huge_val_nan" = "yes"; then
+    AC_DEFINE([HAVE_HUGE_VAL_NAN], 1, [whether HUGE_VAL + -HUGEVAL == NAN])
+  fi
 
-AC_ARG_ENABLE([zend-signals],
-  [AS_HELP_STRING([--disable-zend-signals],
-    [whether to enable zend signal handling])],
-  [ZEND_SIGNALS=$enableval],
-  [ZEND_SIGNALS=yes])
+  dnl Check whether __cpuid_count is available.
+  AC_CACHE_CHECK(whether __cpuid_count is available, ac_cv_cpuid_count_available, [
+    AC_LINK_IFELSE([AC_LANG_PROGRAM([[
+      #include <cpuid.h>
+    ]], [[
+      unsigned eax, ebx, ecx, edx;
+      __cpuid_count(0, 0, eax, ebx, ecx, edx);
+    ]])], [
+      ac_cv_cpuid_count_available=yes
+    ], [
+      ac_cv_cpuid_count_available=no
+    ])
+  ])
+  if test "$ac_cv_cpuid_count_available" = "yes"; then
+    AC_DEFINE([HAVE_CPUID_COUNT], 1, [whether __cpuid_count is available])
+  fi
 
-AC_CHECK_FUNC(sigaction, [
-	AC_DEFINE(HAVE_SIGACTION, 1, [Whether sigaction() is available])
-], [
-	ZEND_SIGNALS=no
-])
-if test "$ZEND_SIGNALS" = "yes"; then
-	AC_DEFINE(ZEND_SIGNALS, 1, [Use zend signal handling])
-	CFLAGS="$CFLAGS -DZEND_SIGNALS"
-fi
+  ZEND_DLSYM_CHECK
 
-AC_MSG_CHECKING(whether to enable zend signal handling)
-AC_MSG_RESULT($ZEND_SIGNALS)
+  AC_ARG_ENABLE([maintainer-zts],
```

Comment:
Having the AC_ARG_ENABLEs be part of ZEND_INIT seems pretty odd to me.

Reply:
So so... For example, with LT_INIT (the libtool) there is also similar approach. What would be here better? Moving this to a separate macro as before (approximately) or moving these to the PHP options macro altogether?

Reply:
I'll add the AC_ARG_ENABLE steps to a separate ZEND_something macro then... Probably better to separate such units of checks like this, yes.